nexus/db-queries/src/db/datastore/sled.rs

```diff
@@ -1159,6 +1380,1067 @@ pub(in crate::db::datastore) mod test {
+        name: &'static str,
+        policy: external::AffinityPolicy,
+    ) -> AntiAffinityGroup {
+        datastore
+            .anti_affinity_group_create(
+                &opctx,
+                &authz_project,
+                AntiAffinityGroup::new(
+                    authz_project.id(),
+                    params::AntiAffinityGroupCreate {
+                        identity: external::IdentityMetadataCreateParams {
+                            name: name.parse().unwrap(),
+                            description: "desc".to_string(),
+                        },
+                        policy,
+                        failure_domain: external::FailureDomain::Sled,
+                    },
+                ),
+            )
+            .await
+            .unwrap()
+    }
+
+    // This short-circuits some of the logic and checks we normally have when
+    // creating affinity groups, but makes testing easier.
+    async fn add_instance_to_anti_affinity_group(
+        datastore: &DataStore,
+        group_id: Uuid,
+        instance_id: Uuid,
+    ) {
+        use db::model::AntiAffinityGroupInstanceMembership;
+        use db::schema::anti_affinity_group_instance_membership::dsl as membership_dsl;
+        use omicron_uuid_kinds::AntiAffinityGroupUuid;
+
+        diesel::insert_into(
+            membership_dsl::anti_affinity_group_instance_membership,
+        )
+        .values(AntiAffinityGroupInstanceMembership::new(
+            AntiAffinityGroupUuid::from_untyped_uuid(group_id),
+            InstanceUuid::from_untyped_uuid(instance_id),
+        ))
+        .on_conflict((membership_dsl::group_id, membership_dsl::instance_id))
+        .do_nothing()
+        .execute_async(&*datastore.pool_connection_for_tests().await.unwrap())
+        .await
+        .unwrap();
+    }
+
+    async fn create_affinity_group(
+        opctx: &OpContext,
+        datastore: &DataStore,
+        authz_project: &authz::Project,
+        name: &'static str,
+        policy: external::AffinityPolicy,
+    ) -> AffinityGroup {
+        datastore
+            .affinity_group_create(
+                &opctx,
+                &authz_project,
+                AffinityGroup::new(
+                    authz_project.id(),
+                    params::AffinityGroupCreate {
+                        identity: external::IdentityMetadataCreateParams {
+                            name: name.parse().unwrap(),
+                            description: "desc".to_string(),
+                        },
+                        policy,
+                        failure_domain: external::FailureDomain::Sled,
+                    },
+                ),
+            )
+            .await
+            .unwrap()
+    }
+
+    // This short-circuits some of the logic and checks we normally have when
+    // creating affinity groups, but makes testing easier.
+    async fn add_instance_to_affinity_group(
+        datastore: &DataStore,
+        group_id: Uuid,
```

Comment:
Nit: All callers look correct, but I find it a bit strange to take untyped uuids next to each other as parameters and then convert them inside the function to their typed equivalents. That sidesteps one of the benefits of typed uuids.

Reply:
Good catch, I'll patch these functions to using typed uuids.